File: CS/2161_Partition_Array_According_to_Given_Pivot.cs
public class Solution {
    public int[] PivotArray(int[] nums, int pivot) {
        var listL = new List<int>();
        var listE = new List<int>();
        var listR = new List<int>();
        for(int i = 0; i < nums.Length; i++){
            if(nums[i] > pivot){
                listR.Add(nums[i]);
            }
            else if(nums[i] < pivot){
                listL.Add(nums[i]);
            }
            else{
                listE.Add(nums[i]);
            }
        }
        listL.AddRange(listE);
        listL.AddRange(listR);
        return listL.ToArray();
        /*
        int[] result = new int[nums.Length];
        int countL = 0;
        int countE = 0;
        int countR = 0;
        for(int i = 0; i < nums.Length; i++){
            if(nums[i] > pivot){
                countR++;
            }
            else if(nums[i] < pivot){
                countL++;
            }
            else{
                countE++;
            }
        }
        
        int indexL = 0;
        int indexE = countL;
        int indexR = countL + countE;
        for(int i = 0; i < nums.Length; i++){
            if(nums[i] > pivot){
                result[indexR] = nums[i];
                indexR++;
            }
            else if(nums[i] < pivot){
                result[indexL] = nums[i];
                indexL++;
            }
            else{
                result[indexE] = nums[i];
                indexE++;
            }
        }
        
        return result;
        */
    }
}

File: Cpp/2161_Partition_Array_According_to_Given_Pivot.cpp
class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        vector<int> result(nums.size());
        int l = 0;
        int e = 0;
        //int m = 0;
        for(int num : nums){
            if(num < pivot){
                l++;
            }
            else if(num == pivot){
                e++;
            }
            //else{
            //    m++;
            //}
        }
        int indexL = 0;
        int indexE = l;
        int indexM = l+e;
        for(int num : nums){
            if(num < pivot){
                result[indexL++] = num;
                //indexL++;
            }
            else if(num == pivot){
                result[indexE++] = num;
                //indexE++;
            }
            else{
                result[indexM++] = num;
                //indexM++;
            }
        }
        
        return result;
    }
};
